engine/src/main/java/org/terasology/rendering/logic/MeshRenderer.java

```diff
@@ -244,6 +242,21 @@ private void renderEntitiesByMaterial(SetMultimap<Material, EntityRef> meshByMat
         }
     }
 
+    /**
+     * Checks whether the entity at the given position is relevant.
+     * <p>
+     * The entity is at the given position is relevant if
+     * a) the entity itself is always relevant, or
+     * b) if the block at the position is relevant.
+     *
+     * @param entity   the entity to check for relevance
+     * @param position the position the entity is located
+     * @return true if the entity itself or the block at the given position are relevant, false otherwise.
+     */
+    private boolean isRelevant(EntityRef entity, Vector3f position) {
+        return entity.isAlwaysRelevant() || worldProvider.isBlockRelevant(position);
```

Comment:
Have you considered what is the most likely case if any? Is it more likely that an entity is always relevant or is it more likely that the block at the given position is relevant? If the latter is more likely you'd gain a smidgen of performance by inverting the two expressions.